libgeotiff/bin/listgeo.c

```diff
@@ -227,20 +226,29 @@ static void GTIFPrintCorners( GTIF *gtif, GTIFDefn *defn, FILE * fp_out,
 
 {
     printf( "\nCorner Coordinates:\n" );
+
+    unsigned short raster_type = RasterPixelIsArea;
+    GTIFKeyGet(gtif, GTRasterTypeGeoKey, &raster_type, 0, 1);
```

Comment:
I am not entirely pleased with this solution to obtain the value of the key, as it ignores the return value and doesn't check for any undesired values in raster_type. Should one be strict here?

Reply:
Can you use GTIFKeyGetSHORT() instead of GTIFKeyGet() ? This is safer in case of corrupted files.
Your current logic otherwise looks reasonable to me.

Reply:
> Can you use GTIFKeyGetSHORT() instead of GTIFKeyGet() ?

Unfortunately GTIFKeyGetSHORT and related functionality is declared static in geo_normalize.c, so not without further changes.

Reply:
Good point. I think it could be worth exporting GTIFKeyGetSHORT() and GTIFKeyGetDOUBLE() in geotiff.h

Reply:
Sounds like a good idea, but should probably go into another PR?

Reply:
I'm fine if it goes into that PR as a separate commit. By the way if you rebase your PR on top of latest master, the CI failure on Mac should now be resolved

Reply:
Rebased, will look into exposing the mentioned functionality next.